qiskit/circuit/library/standard_gates/__init__.py

```diff
@@ -107,14 +107,20 @@
 
 def get_standard_gate_name_mapping():
     """Return a dictionary mapping the name of standard gates and instructions to an object for
-    that name."""
+    that name.
+
+    For variadic operations (i.e. operations that operate on a variable
+    number of qubits), such as :class:`~.MCXGate`, the class for that
+    operation is returned and not an instance.
+    """
     from qiskit.circuit.parameter import Parameter
     from qiskit.circuit.measure import Measure
     from qiskit.circuit.delay import Delay
     from qiskit.circuit.reset import Reset
+    from qiskit.circuit.controlflow import IfElseOp, WhileLoop, ForLoop
```

Comment:
I'm not really certain we should be returning these as "standard gates"; they're no gates, and they don't really follow a standard form of accepting n float-like parameters.

I suppose it depends a bit on how the result of this function is meant to be used.

Reply:
Yeah, I was on the fence about this too, but we already had `Delay`, `Reset`, etc here. This was added in #8759 so we had a mapping for `basis_gate` entries to object for a target. I guess the function name isn't great and we should rename it (well, alias, then deprecate, then rename).